src/pages/search.js

```diff
@@ -40,12 +52,87 @@ export default withSearch(({ search }) => {
     ? ` : ${query} (${totalHits} result${totalHits === 1 ? '' : 's'})`
     : ''
 
-  let searchEvent
+  const getSlug = item => {
+    if (!item.publishDate) {
```

Comment:
I think it's risky to rely on `item.publishDate`, imo adding a `type` argument to the arrow function would be safer

Reply:
Yeah, you're right, that's brittle. The type is not known when `getSlug` is called, though. We could refactor it to loop over those hits arrays separately and set the type there, I think. That could get a little messy as well, but I'll see if I can come up with something that will work.

Reply:
Imo looping over the results while retaining the type is the way to go, added `searchResultTypes` const to make it easier and gonna make result keys use keys from this constant so it should be more robust

Reply:
Used your new keys and `getSanitizedSlug` function to improve this.